R/create_burden_templates.R

```diff
@@ -0,0 +1,537 @@
+##' Creat burden estimate templates
+##' @title Creat burden estimate templates
+##'
+##' @param files filenames, it has to contain the following three files
+##' "Country_Disease_Table.csv", "model_meta.csv" and "model_outcomes.csv"
+##'
+##' @param files_path the directory that contains those files above
+##'
+##' @param assert_files needed meta files
+##' "Country_Disease_Table.csv", "model_meta.csv" and "model_outcomes.csv"
+##'
+##' @param central_templates_only TRUE if you only need central burden templates, usually for open call. But it depend on Tini's input.
+##'  Set to FALSE as default, which outputs both central and stochastic templates
+##'
+##' @export
+create_burden_template <- function(files, files_path, assert_files = c("Country_Disease_Table.csv", "model_meta.csv", "model_outcomes.csv"), central_templates_only = FALSE) {
+  assert_has_files(files, assert_files)
+  files <- paste0(files_path, files)
+
+  country_disease_table <- read_csv(files[1])
+  model_meta_data <- read_csv(files[2])
+  model_outcomes <- read_csv(files[3])
+
+  names(country_disease_table)[names(country_disease_table) == "Country"] <-
+    "country_name"
+  names(country_disease_table)[names(country_disease_table) == "ISO3.code"] <-
+    "country"
+
+  ## making sure that the final year makes sense:
+  year_end <- 2030
+  model_meta_data$year_max <- apply(cbind(2100, year_end + model_meta_data$burden_max_age), 1, min)
+  ## write templates
+  output_files <- sapply(seq_along(model_meta_data[, 1]),
+                         function(i) write_template_files(model_meta_data[i, ],
+                                                          year_end,
+                                                          country_disease_table,
+                                                          model_outcomes,
+                                                          central_templates_only))
+}
+
+### From Nick's insert_function.R
+# function adds new rows to dataframe with contents of value_insert
+# to the column column_name
+
+insert_dataframe <- function(data,column_name,value_insert) {
```

Comment:
Can you explain to me this function please?  I'm quite confused about what it is doing - there are more operations in here than I'd expect